components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -1158,3 +1216,22 @@ internal fun EngineSession.LoadUrlFlags.getGeckoFlags(): Int = if (contains(ALLO
 } else {
     value
 }
+
+/**
+ * Create a Content Disposition formatted string with the receiver used as the filename and
+ * file extension set as PDF.
+ */
+internal fun String.pdfContentDisposition() = take(MAX_TITLE_LENGTH).run {
+    "attachment; filename=$this.pdf;"
+}
+
+/**
+ * Maximum number of characters for the title length. This is used for limiting the length of a
+ * filename of a PDF that is generated from the session.
+ */
+private const val MAX_TITLE_LENGTH = 500
```

Comment:
This feels like a potentially long filename length for the filesystem to me, but I couldn't find any official documentation stating Android filename length limits. I suppose the final saving mechanism will check for things like this too?

Reply:
Ha, that was the placeholder value we used and I didn't consider it again after that.

> official documentation stating Android filename length limits.

Since Android is linux based, I'd assume the core limitations would come from there. A quick search gave me this wiki page via stackoverflow: https://en.wikipedia.org/wiki/Comparison_of_file_systems#Limits

This isn't super helpful unless we know what the filesystem is. I know Android has used ext4 for the fs, but I did another search to see what other variants or version would be expected and found this: https://knowtechie.com/which-file-systems-can-android-read/

> Android has always supported the FAT32, Ext3, and Ext4 file system formats, but external drives are often formatted in exFAT or NTFS

So that means we should be limited to.. 255 bytes!

Will update the PR with that value and a short description about this magic number.